index.js

```diff
@@ -35,8 +41,18 @@ module.exports = function (){
     }
 
     this.getWeb3 = async function (contractABI,contractAddress,provider){
+        try{
+            web3.setProvider(new web3.providers.HttpProvider(provider));
+            this.ContractInstance = await new web3.eth.Contract(contractABI,contractAddress);
+            return web3;
+        }
+        catch (err){
+            return null;
+        }
+    }
+
+    this.getWeb3Base = function (provider){
```

Comment:
This works, but it is against the singelton approach

Reply:
Singleton would limit possibilities of web3js-raw